admission_control/admission-control-service/src/admission_control_service.rs

```diff
@@ -89,28 +89,27 @@ where
         if !self.can_send_txn_to_mempool()? {
             debug!("Mempool is full");
             OP_COUNTERS.inc_by("submit_txn.rejected.mempool_full", 1);
-            let mut response = SubmitTransactionResponse::new();
-            let mut status = MempoolAddTransactionStatus::new();
+            let mut response = SubmitTransactionResponse::default();
+            let mut status = MempoolAddTransactionStatus::default();
             status.set_code(MempoolIsFull);
-            status.set_message("Mempool is full".to_string());
-            response.set_mempool_status(status);
+            status.message = "Mempool is full".to_string();
+            response.status = Some(Status::MempoolStatus(status));
             return Ok(response);
         }
 
-        let signed_txn_proto = req.get_signed_txn();
+        let signed_txn_proto = req.signed_txn.clone().unwrap_or_else(Default::default);
```

Comment:
don't you want to just error if you can't unwrap here?

Reply:
In some other places I error'ed out instead of us default. With the old protobuf generation code you would call `.get_<field>` and if the field was `None` it would give you a default instead of erroring so really i was trying to preserve the existing functionality. Definitely a place we could look closer at